src/contracts/core/DelegationManager.sol

```diff
@@ -685,6 +699,26 @@ contract DelegationManager is
         _beaconChainSlashingFactor[staker] = bsf;
     }
 
+    /// @dev Calculate amount of withdrawable shares from an operator for a strategy that is still in the queue
+    /// and therefore slashable.
+    function _getSlashableSharesInQueue(address operator, IStrategy strategy) internal view returns (uint256) {
+        uint256 currCumulativeWithdrawalShares = uint256(_cumulativeWithdrawalsHistory[operator][strategy].latest());
+        // Note: this will simply return 0 if no history exists, same for latest() as well
+        uint256 pastCumulativeWithdrawalShares = _cumulativeWithdrawalsHistory[operator][strategy]
+            .getAtProbablyRecentBlock(block.number - MIN_WITHDRAWAL_DELAY_BLOCKS);
+        return currCumulativeWithdrawalShares - pastCumulativeWithdrawalShares;
+    }
+
+    /// @dev Update the cumulative withdrawn shares from an operator for a given strategy
+    function _updateCumulativeWithdrawnShares(
+        address operator,
+        IStrategy strategy,
+        uint256 queueWithdrawnShares
+    ) internal {
+        uint256 currCumulativeWithdrawalShares = uint256(_cumulativeWithdrawalsHistory[operator][strategy].latest());
```

Comment:
this PR looks nice!

my main feedback is that we should store _cumulative scaled shares_ queued for withdrawal, not shares.

Upon burning we should multiply the scaled shares in the withdrawal queue by the slashed magnitude. This would make it so that view functions for introspection into the withdrawal queue automatically update to reflect correct values after slashing

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/881/commits/138eef799d1ce06c60ca1f2b06b626ff2410dfa5 
Added, had to change calcSlashedAmount back to old interface but it is more consistent with the way scaledShares are also calculated.